cocos/core/root.ts

```diff
@@ -605,12 +605,16 @@ export class Root {
                 case LightType.SPOT:
                     l.scene.removeSpotLight(l as SpotLight);
                     break;
+                case LightType.DIRECTIONAL:
+                    l.scene.removeDirectionalLight(l as DirectionalLight);
+                    break;
                 default:
                     break;
                 }
+                p.free(l);
+                l.destroy();
```

Comment:
If light is destroyed, it shouldn't be free to the light pool, otherwise unexpected behavior will occur when it's reused

Reply:
1. `destroyLight` should only destroy it, shouldn't free it
2. We can add a `recycleLight` function to free it but not destroy it
3. All light component's `onDestroy` should only invoke `recycleLight` so that the light object will be reused

Reply:
sure，I will change that on the v3.6.0 version

Reply:
v3.5.2 won't accept this PR, please directly rebase the most suitable modification to v3.6